mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java

```diff
@@ -398,14 +396,24 @@ public void onConfigurationChanged(Configuration newConfig) {
     @Override
     public void onResume() {
         Log.d(TAG, "onResume()");
-
         super.onResume();
+
         ConnectionFactory.addListener(this);
         MemorizingTrustManager.getInstance(this).bindDisplayActivity(this);
 
         onAvailableConnectionChanged();
         updateNotificationDrawerItem();
 
+        if (mInitState == InitState.DONE &&
```

Comment:
Another approach that didn't occur to me earlier:
- Add an intent extra to the settings activity's result intent that flags a cleared sitemap (like for theme change)
- Check that flag in onActivityResult (check for sitemap emptiness not needed anymore at that point)

Reply:
Yes, this is a lot cleaner.